infrastructure/main.tf

```diff
@@ -26,3 +26,9 @@ module "database" {
   physical_location   = module.resource_group.physical_location
   my_subnet_id        = module.vnet.my_subnet_id
 }
+
+module "blob_storage" {
+  source = "./modules/blob_storage"
+  resource_group_name = module.resource_group.resource_group_name
+  physical_location = module.resource_group.physical_location
+}
```

Comment:
You declared in your ./blob/variables.tf that the module is waiting for 4 variables : rg_name, physical_location, storage_container_name and storage_blob_name. You need here to give your module these 4 vars. Actually you only give 2 of them